cub/cub/thread/thread_operators.cuh

```diff
@@ -414,121 +467,283 @@ _CCCL_HOST_DEVICE BinaryFlip<BinaryOpT> MakeBinaryFlip(BinaryOpT binary_op)
   return BinaryFlip<BinaryOpT>(binary_op);
 }
 
+/***********************************************************************************************************************
+ * SIMD Operators
+ **********************************************************************************************************************/
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
+
+// TODO: extend to floating_point<M, E>
+
 namespace internal
 {
-// TODO: Remove DPX specilization when nvbug 4823237 is fixed
 
 template <typename T>
-struct DpxMin
+struct SimdMin
 {
-  static_assert(detail::always_false<T>(), "DpxMin is not supported for this type");
+  static_assert(cub::detail::always_false<T>(), "Unsupported specialization");
 };
 
 template <>
-struct DpxMin<::cuda::std::int16_t>
+struct SimdMin<::cuda::std::int16_t>
 {
+  using simd_type = unsigned;
+
   _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE unsigned operator()(unsigned a, unsigned b) const
```

Comment:
I am wondering whether this will emit warnings because of the signed unsigned difference.

Also i am generally wondering why we dont take the actual type and convert within the function implementation.

That would potentially allow for a massive amount fo simplifications

Reply:
there is a top-level `bit_cast` from two `int16_t/uint16_t` to a single `unsigned